optimade/server/entry_collections.py

```diff
@@ -104,6 +107,16 @@ def __init__(
             version=(0, 10, 1), variant="default"
         )  # The MongoTransformer only supports v0.10.1 as the latest grammar
 
+        # check aliases do not clash with mongo operators
+        _all_aliases = self.resource_mapper.all_aliases()
+        if any(
+            alias[0].startswith("$") or alias[1].startswith("$")
+            for alias in _all_aliases
+        ):
+            raise RuntimeError(
+                f"Cannot define an alias starting with a '$': {_all_aliases}"
```

Comment:
```suggestion
                f"Cannot define an alias starting with a '$': {self.resource_mapper.all_aliases()}"
```

Reply:
Another way to go about this (since this method is being used elsewhere in the class, is to cache it as a private instance variable (instead of a the local one you have now).

Reply:
I've cached it as a private instance variable to avoid calling this all over the place